spec/std/array_spec.cr

```diff
@@ -2042,17 +2042,76 @@ describe "Array" do
     end
   end
 
-  it "#resize" do
-    ary = [1, 2, 3]
-    ary.resize(3).should eq([1, 2, 3])
-    ary.resize(4).should eq([1, 2, 3])
-    ary.remaining_capacity.should eq(4)
-    ary.shift
-    ary.resize(3).should eq([2, 3])
-    ary.remaining_capacity.should eq(2)
-
-    expect_raises(ArgumentError, "Insufficient capacity: 2 cannot hold 2 elements with an offset of 1") do
-      ary.resize(2)
+  describe "#ensure_capacity" do
+    it "does nothing if fits" do
+      ary = [1, 2, 3]
+      capacity1 = ary.remaining_capacity
+      ary.ensure_capacity(3)
+      ary.should eq([1, 2, 3])
+      ary.remaining_capacity.should eq(capacity1)
+    end
+
+    it "grows the array if it doesn't fit" do
+      ary = [1, 2, 3]
+      ary.ensure_capacity(4).should eq([1, 2, 3])
+      ary.remaining_capacity.should eq(4)
+    end
+
+    it "rewinds by default the array" do
+      ary = [1, 2, 3]
+      ary.shift
+      ary.ensure_capacity(3).should eq([2, 3])
+      ary.remaining_capacity.should eq(3)
+    end
+
+    it "doesn't rewind when asked not to" do
+      ary = [1, 2, 3]
+      ary.shift
+      ary.ensure_capacity(3, false).should eq([2, 3])
+      ary.remaining_capacity.should eq(2)
+    end
+
+    it "raises if not enough capacity" do
+      expect_raises(ArgumentError, "Insufficient capacity: 2 cannot hold 3 elements") do
+        ary = [1, 2, 3]
+        ary.ensure_capacity(2)
```

Comment:
Again (https://github.com/crystal-lang/crystal/pull/11485#discussion_r755521471) this seems very unexpected: Why would this method raise? We ask to ensure the array has at least a capacity of 2, and the actual capacity is 3. So everything should be good, nothing to do.

Reply:
Agh, sorry I missed this in the original comment. Will fix.